tests/Tests.ServiceProvider/CustomAutoDataAttributes.cs

```diff
@@ -0,0 +1,39 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Tests;
+
+[AttributeUsage(AttributeTargets.Method)]
+public class CustomAutoDataAttribute : AutoDataAttribute
+{
+    public CustomAutoDataAttribute() : base(FixtureHelpers.CreateFixture)
+    {
+    }
+}
+
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+public class CustomInlineAutoDataAttribute : InlineAutoDataAttribute
+{
+    public CustomInlineAutoDataAttribute(params object[] args) : base(FixtureHelpers.CreateFixture, args)
+    {
+    }
+}
+
+internal static class FixtureHelpers
+{
+    public static IFixture CreateFixture()
+    {
+        var fixture = new Fixture();
+
+        fixture.Customize(new AutoMoqCustomization
+        {
+            ConfigureMembers = true,
+            GenerateDelegates = true
+        });
+        
+        fixture.Register<IServiceCollection>(() => new ServiceCollection());
```

Comment:
I would write these as a specimen builders. The `Register` method does a lot of things that eventually boil down to the same behavior, with the drawback that you have to maintain the constructor invocation by hand. Though in your scenario the `ServiceCollection` constructors are unlikely to change so it's fine to use a factory function instead.
```cs
fixture.Customizations.Add(new TypeRelay(typeof(IServiceCollection),typeof(ServiceCollection)));
```